mantidproject/mantid
# Add a selection of tests to the MainWindow class

Adds a selection of unittests to the mainwindow class including
- Check menus look correct
- Check cpp and python interfaces are discovered
- Check LogMessageDisplay captures stdout/stderr
- Closing behaviour

**To test:**
- Ensure that all the tests pass (including in `writetosignal.py`)
- Test everything still appears in the workbench messages pane.

Fixes #28542 





---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
